### ISSUE TITLE
request for binary md5 buf

### ISSUE THREAD

**Issue by MARTIMM** [cosimo/perl6-digest-md5#9]
I would like to have a 16 byte Buf array for md5 result instead of 32 character hex string from md5_hex. It is easy to add a method something like.
```
    multi method md5-buf(Str $str --> Buf) {
        return md5( $str.encode('ascii') );
    }
```
I want this to use it for the BSON binary type with md5 as a subtype although I am not shure yet if the BSON spec demands a 32 character string or a 16 byte array. I would say the last because it is specified under the binary BSON type.

At the moment I need to create this from the 32 character string returned from md5_hex which will have overhead from md5_hex() making a string and from me to convert it back again.

Thanks in advance
Marcel Timmerman

**Comment by FROGGS** [comment]
If you make a pull request I could merge it right away...
Please add your changes to both `MD5.pm` and `MD5.pm.perl`.

And perhaps call it `md5_buf`.

**Comment by MARTIMM** [comment]
Can do that but takes some time caused by other 'worries'

**Comment by FROGGS** [comment]
Got implemented via #10 

Thank you.
